### PR TITLE
service: Store and expose service name and namespace

### PR DESCRIPTION
**Pull request by gandro**
Extends the user-space `service.Service` map with the service name and namespace. The goal is to expose these fields in the Cilium API such that external consumers can obtain the name of each service without having to consult other data sources. The BPF maps are left untouched, as they don't need to store these.

The name and namespace are both optional and are provided by the k8s watcher, though users of the Cilium API may also optionally provide them via `PUT /service/{id}`.

This PR also performs some minor clean-ups on the tests in the `service` package.

Can be reviewed per commit.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9554)

### CODE REVIEW

**Comment by gandro** [comment]
test-me-please

File: api/v1/openapi.yaml [hunk 1]
```diff
@@ -1978,6 +1978,12 @@ definitions:
             enum:
             - ClusterIP
             - NodePort
+          name:
+            description: Service name
+            type: string
+          namespace:
+            description: Service namespace
```
Comment (ianvernon) [inline api/v1/openapi.yaml]: Non-blocking: perhaps mention that these usually correspond to a Kubernetes service name / namespace in the descriptions?

**Review by ianvernon — APPROVED** [review]

**Review by tgraf — APPROVED** [review]

**Comment by gandro** [comment]
test-me-please

File: api/v1/openapi.yaml (hunk 1 above)
Comment (gandro) [reply api/v1/openapi.yaml]: Good point!

**Pull request merged**